doc/tutorial/single_cell_bluepyopt.rst

```diff
@@ -0,0 +1,288 @@
+   axon_parent = morpho.segment_tree.parents[axon_root]
+
+   # load replacement morphology
+   axon_replacement = arbor.load_component(
+      cell_json['morphology']['replace_axon']).component
+   axon_replacement_st = replacement.to_segment_tree()
+
+   # pruned segment tree with axon replacement
+   joined_st = pruned_st.join_at(
+      axon_parent, axon_replacement_st)
+
+   # final morphology
+   morpho_axon_replaced = arbor.morphology(joined_st)
+
+The steps presented here are included when loading an exported model with the above ``read_acc`` function from BluePyOpt.
+
+
+Define locations in the GUI
+---------------------------
+
+BluePyOpt makes use of the Neuron simulator's features to define custom locations for stimuli and voltage probes that can be challenging to translate to Arbor. As an example, the layer-5 pyramidal cell optimization problem has multiple voltage probes located distally on the apical dendrite at a fixed distance from the soma as described in `config/protocols.json <https://github.com/BlueBrain/BluePyOpt/blob/master/examples/l5pc/config/protocols.json>`_.
+
+To describe these locations accurately in Arbor, we will use the Arbor GUI. For this purpose, we create an :class:`arbor.cable_cell` and export it to ACC so that it can be loaded in the GUI,
+
+.. code-block:: python
+
+   cell = arbor.cable_cell(morpho, labels, decor)
+   arbor.write_component(cell, 'path/to/l5pc_cable_cell.acc')
+
+After loading this in file in the GUI with ``File > Cable cell > Load``, we can click the ``Locations`` and ``Cell`` tab to obtain the following overview.
+
+.. figure:: single_cell_bluepyopt_gui_l5pc_overview.png
+    :width: 800
+    :align: center
+
+As a first attempt to formulate the location at 660 µm from the soma, we can use
+
+.. code-block::
+
+   (distal-translate (on-components 0.5 (region "soma")) 660)
+
+and since we're only interested in the apical dendrite, we can wrap this in ``(restrict ... (region "apic"))``.
+
+Unfortunately, for this particular morphology, this does not result in a concrete location as can be seen in the Arbor GUI below (Locset A, violet, is empty). Alternatively, the method :func:`arbor.cable_cell.locations` will also return an empty list of concrete locations for this expression. To obtain a valid location instead, we can replace ``(on-components 0.5 (region "soma"))`` by the proximal boundary of the apical dendrite with ``(proximal (region "apic"))`` (Locset B, orange). The restrict operation to the apical region is no longer necessary. But now, the resulting concrete location is non-unique.
+
+.. figure:: single_cell_bluepyopt_gui_l5pc_bAP_attempt.png
+    :width: 800
+    :align: center
+
+BluePyOpt solves this issue with the Neuron simulator by choosing a particular section on the apical dendrite. While a similar approach is possible in Arbor using the :func:`arbor.cable_cell.locations` method, for reproducibility it is preferable to determine the location directly in the Arbor GUI.
```

Comment:
`BluePyOpt solves this issue with the Neuron simulator by choosing a particular section on the apical dendrite.`

This isn't clear. Do you mean that Neuron chooses one at random?

Reply:
The function [find_icomp](https://github.com/BlueBrain/BluePyOpt/blob/master/bluepyopt/ephys/locations.py#L284) just picks the last location with non-vanishing diameter at given distance from the soma after iterating over all of them. So Neuron decides on the order, BluePyOpt picks the last one. The advantage of the GUI procedure is that you have visual control over what probe location you select (and you're sure the S-expression really evaluates to what you believe it should).

Reply:
Also, I didn't check the voltages at all the locations shown in the second figure. But not all of them are 660 um away from the soma on the high apical branches, some are just terminals much lower on the apical dendrite. For the latter, I'd expect the traces to look differently.

Reply:
I'll emphasize the user-aspect more. There's an [explicit check](https://github.com/lukasgd/BluePyOpt/blob/9443b123c72f35adfba747114406410b4ff3c160/bluepyopt/ephys/protocols.py#L604-L610) in the Arbor protocol that enforces a unique recording (probe) location, which is by design what BluePyOpt expects. If the user decides in the GUI which location to choose, I wouldn't call that picking at random, although, of course the user is free to do that.

With Neuron, the implementation in BluePyOpt just chooses the last section with non-vanishing diameter at that fixed distance from the soma, which depends on Neuron's internal section ordering that is probably not random but neither something guaranteed to be fixed at the API-level.